### PR TITLE
DEPR: Series.astype(np.datetime64)

### LINKED ISSUE #47844: DOCS: series.astype return error when a datetime64 (unitless) series astype numpy.datetime64
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
pandas master branch:

import pandas as pd
import numpy
>>> pd.Series(['1970-01-01', '1970-01-01', '1970-01-01'], dtype="datetime64[ns]").astype(numpy.datetime64, copy=False)
*** TypeError: Cannot cast DatetimeArray to dtype datetime64

```

### Issue Description

I'm not sure whether it is expect behavior. I can repro on main branch, but can not repro on 1.4.x.
I know pandas community introduce a new feature about [ENH: DTI/DTA.astype support non-nano](https://github.com/pandas-dev/pandas/commit/67e8c4c3761ab1da4b0a341a472c0fe2ea393e8b) . But the new feature disable the original code tree.
In this case, the original code tree will locate on 
```
        elif self.tz is None and is_datetime64_dtype(dtype) and dtype != self.dtype:
            # unit conversion e.g. datetime64[s]
            return self._ndarray.astype(dtype)
```

But now, it is
```
        return dtl.DatetimeLikeArrayMixin.astype(self, dtype, copy)
```
And raise an error.

### Expected Behavior

```
>>> import pandas as pd
>>> import numpy
>>> pd.Series(['1970-01-01', '1970-01-01', '1970-01-01'], dtype="datetime64[ns]").astype(numpy.datetime64, copy=False)
0   1970-01-01
1   1970-01-01
2   1970-01-01
dtype: datetime64[ns]
```

### Installed Versions

<details>

pandas-1.5.0.dev0+1135.gc711be009b

</details>

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #47844 (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -1086,6 +1087,11 @@ def astype(self, dtype, copy: bool = True):
 
         values = self._data
         if isinstance(values, ExtensionArray):
+            if isinstance(dtype, np.dtype) and dtype.kind == "M" and is_unitless(dtype):
```
Comment (mroeschke) [inline pandas/core/indexes/base.py:1090]: Is this for users who may have created a custom EA w/ dtype with these conditions? Would it need a deprecation as well?

Comment (jbrockmendel) [reply pandas/core/indexes/base.py:1090]: this is to keep the 1.4.x behavior for Index

Comment (mroeschke) [reply pandas/core/indexes/base.py:1090]: Got it, so we have a test that covers this case?

Comment (jbrockmendel) [reply pandas/core/indexes/base.py:1090]: yes

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**